mmv1/products/vertexai/IndexEndpointDeployedIndex.yaml

```diff
@@ -0,0 +1,264 @@
+  - !ruby/object:Api::Type::NestedObject
+    name: privateEndpoints
+    output: true
+    description: Provides paths for users to send requests directly to the deployed index services running on Cloud via private services access
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'matchGrpcAddress'
+        description: The ip address used to send match gRPC requests.
+        output: true
+      - !ruby/object:Api::Type::String
+        name: 'serviceAttachment'
+        description: The name of the service attachment resource. Populated if private service connect is enabled.
+        output: true
+      - !ruby/object:Api::Type::Array
+        name: 'pscAutomatedEndpoints'
+        description: |
+          PscAutomatedEndpoints is populated if private service connect is enabled if PscAutomatedConfig is set.
+        output: true
+        item_type: !ruby/object:Api::Type::NestedObject
+          properties:
+            - !ruby/object:Api::Type::String
+              name: projectId
+              output: true
+              description: |
+                Corresponding projectId in pscAutomationConfigs
+            - !ruby/object:Api::Type::String
+              name: network
+              output: true
+              description: |
+                Corresponding network in pscAutomationConfigs.
+            - !ruby/object:Api::Type::String
+              name: 'matchAddress'
+              output: true
+              description: |
+                ip Address created by the automated forwarding rule.
+  - !ruby/object:Api::Type::String
+    name: 'indexSyncTime'
+    output: true
+    description: If this timestamp's value is at least the Index.update_time of the original Index, it means that this DeployedIndex and the original Index are in sync. If this timestamp is older, then to see which updates this DeployedIndex already contains (and which it does not), one must list the operations that are running on the original Index.
+  - !ruby/object:Api::Type::NestedObject
+    name: automaticResources
+    description: A description of resources that the DeployedIndex uses, which to large degree are decided by Vertex AI, and optionally allows only a modest additional configuration.
+    default_from_api: true
+    properties:
+      - !ruby/object:Api::Type::Integer
+        name: 'minReplicaCount'
+        description: The minimum number of replicas this DeployedModel will be always deployed on.
+        default_from_api: true
+      - !ruby/object:Api::Type::Integer
+        name: 'maxReplicaCount'
+        description: The maximum number of replicas this DeployedModel may be deployed on when the traffic against it increases. If maxReplicaCount is not set, the default value is minReplicaCount
+        default_from_api: true
+  - !ruby/object:Api::Type::NestedObject
+    name: dedicatedResources
```

Comment:
There's a nested field called autoscalingMetricSpecs [described in the docs](https://cloud.google.com/vertex-ai/docs/reference/rest/v1/DedicatedResources) that isn't implemented here- is that intentional or an oversight?

Reply:
Refer to https://github.com/GoogleCloudPlatform/magic-modules/pull/11039#discussion_r1700382022